File: refactoring-practice-1/src/main/java/dojo/introduce_null_object/Customer.java
package dojo.introduce_null_object;

public class Customer {

	private final String name;
	private final BillingPlan billingPlan;
	private final BillingHistory billingHistory = new BillingHistory();
	
	public Customer(String name, BillingPlan billingPlan) {
		super();
		this.name = name;
		this.billingPlan = billingPlan;
	}
	
	public String getName() {
		return name;
	}
	
	public BillingPlan getBillingPlan() {
		return billingPlan;
	}
	
	public BillingHistory getBillingHistory() {
		return billingHistory;
	}
}

File: refactoring-practice-1/src/main/typescript/introduce_null_object/customer.ts
import {BillingPlan} from "./billing-plan.js";
import {BillingHistory} from "./billing-history.js";

export class Customer {

    private billingHistory: BillingHistory = new BillingHistory();

    public constructor(private name: string, private billingPlan: BillingPlan) {
    }

    public getName() {
        return this.name;
    }

    public getBillingPlan(): BillingPlan {
        return this.billingPlan;
    }

    public getBillingHistory(): BillingHistory {
        return this.billingHistory;
    }
}
